test/plugin/allow_assign_current_test.rb

```diff
@@ -0,0 +1,49 @@
+require "test_helper"
+require "shrine/plugins/allow_assign_current"
+
+describe Shrine::Plugins::AllowAssignCurrent do
+  before do
+    @attacher = attacher { plugin :allow_assign_current }
+    @shrine   = @attacher.shrine_class
+  end
+
+  it "ignores assigning current non-cached file" do
+    file = @attacher.upload(fakeio, :other_store)
+    @attacher.file = file
+    @attacher.assign(file.to_json)
+    assert_equal file , @attacher.file
```

Comment:
This test shows the file is the same but not sure if it tests whether it was due to it being skipped or uploading the same file. Same for test on L24 below.

Reply:
This test asserts that no exception is raised, while the test below asserts that metadata is not changing. These two tests together should assert we're "skipping" the assignment (along with the `restore_cached_data` test at the bottom). If there is some other behavior you think should be asserted, let me know.